[ET-VK][Ops] dequantize_per_tensor.default test setup

ahmtox opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #11553
* #11557
* #11483
* #11369
* #11556
* #11555
* #11554
* #11482
* __->__ #11481
* #11480
* #11368
* #11367
* #11366
* #11575
* #11365
* #11552
* #11479

Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

Commit:
[ET-VK][Ops] dequantize_per_tensor.default test setup

Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_tensor.default test setup"


Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_tensor.default test setup"


Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]

manuelcandales commented on backends/vulkan/test/op_tests/dequantize_test.cpp:
```diff
@@ -314,3 +314,337 @@ void check_dequantize_args(
+  ASSERT_TRUE(output_correct);
+}
+
+// Test cases for dequantize_per_tensor
```
also add tests for int32 and int64

ahmtox replied:
theres no real support for int64 in the shader. right now dequantize matches the IO for quantize which include half, float and then uint8, int8, and int32 as common dtypes and to avoid generating excess variants. if int64 is a necessary variant I'll make those changes in a future diff.

there is a [diff](https://github.com/pytorch/executorch/pull/11553) that adds double support (though really just fp32 underneath)

i will add more tests in this diff.

SS-JIA replied:
As @ahmtox has pointed out, to reduce binary size Vulkan kind of takes the opposite approach of the portable kernels, which means supporting the *minimum* dtypes necessary i.e. basically what we would expect to encounter in a "real" model.

Commit:
Update on "[ET-VK][Ops] dequantize_per_tensor.default test setup"


Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_tensor.default test setup"


Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_tensor.default test setup"


Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_tensor.default test setup"


Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_tensor.default test setup"


Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_tensor.default test setup"


Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_tensor.default test setup"


Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_tensor.default test setup"


Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_tensor.default test setup"


Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]

Commit:
Update on "[ET-VK][Ops] dequantize_per_tensor.default test setup"


Creating dequantize_per_tensor testing framework along with a reference implementation for testing

Differential Revision: [D76267054](https://our.internmc.facebook.com/intern/diff/D76267054/)

[ghstack-poisoned]
